dropshot/src/server.rs

```diff
@@ -901,11 +902,24 @@ async fn http_request_handle<C: ServerContext>(
             // For CancelOnDisconnect, we run the request handler directly: if
             // the client disconnects, we will be cancelled, and therefore this
             // future will too.
-            //
-            // TODO-robustness: We should log a warning if we are dropped before
-            // this handler completes; see
-            // https://github.com/oxidecomputer/dropshot/pull/701#pullrequestreview-1480426914.
-            handler.handle_request(rqctx, request).await?
+            // In case it is indeed cancelled, we keep track of the handler
+            // progress with `done`, so the scopeguard below can log a warning
+            // if it's dropped before completion.
+            let done = AtomicBool::new(false);
+            let request_log = rqctx.log.clone();
```

Comment:
Since this is common with L928, you could move it up to L899 and delete this one and the one on L928.